ext/notify/pagerduty/pagerdutynotifier_test.go

```diff
@@ -0,0 +1,79 @@
+package pagerduty
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/odpf/optimus/models"
+)
+
+type PagerDutyServiceImplMock struct {
+	mock.Mock
+}
+
+func (s *PagerDutyServiceImplMock) SendPagerDutyAlert(ctx context.Context, evt Event) error {
+	s.Called(ctx, evt)
+	return nil
+}
+
+func TestPagerDuty(t *testing.T) {
+	t.Run("should send alert to pagerduty service using service name successfully", func(t *testing.T) {
+		ctx, cancel := context.WithCancel(context.Background())
+		var sendErrors []error
+
+		pagerDutyServiceMock := new(PagerDutyServiceImplMock)
+		pagerDutyServiceMock.On("SendPagerDutyAlert", ctx,
```

Comment:
Please assert expectation on the mock, like `defer pagerDutyServiceMock.AssertExpectations(t)` after this line.